backend/python/app/services/implementations/story_service.py

```diff
@@ -113,25 +113,41 @@ def create_translation(self, translation):
             raise error
         return new_story_translation
 
-    def get_story_translations(self, user_id, is_translator, language, level):
+    def get_story_translations_by_user(self, user_id, is_translator, language, level):
+        if is_translator is None:
+            role_filter = (StoryTranslation.translator_id == user_id) | (
+                StoryTranslation.reviewer_id == user_id
+            )
+        elif is_translator:
+            role_filter = StoryTranslation.translator_id == user_id
+        else:
+            role_filter = StoryTranslation.reviewer_id == user_id
+        return self.get_story_translations(
+            language=language, level=level, role_filter=role_filter
+        )
+
+    def get_story_translations(
+        self, language=None, level=None, stage=None, story_title=None, role_filter=None
+    ):
         try:
-            if is_translator is None:
-                role_filter = (StoryTranslation.translator_id == user_id) | (
-                    StoryTranslation.reviewer_id == user_id
-                )
-            elif is_translator:
-                role_filter = StoryTranslation.translator_id == user_id
-            else:
-                role_filter = StoryTranslation.reviewer_id == user_id
+            filters = []
+            if role_filter is not None:
+                filters.append(role_filter)
+            if language is not None:
+                filters.append(StoryTranslation.language == language)
+            if stage is not None:
+                filters.append(StoryTranslation.stage == stage)
+            if level is not None:
+                filters.append(Story.level == level)
+            if story_title is not None:
+                filters.append(Story.title.like(f"%{story_title}%"))
```

Comment:
I think what might be more intuitive is `"{story_title}%"` since a short `story_title` (eg `a`) might have too many hits. 
However, I also see that if we delay the api call on the frontend (such that atleast a few characters are entered), the current method is advantageous. eg `story_title="gatsby"` will still return `"The Great Gatsby"`. What do you think?

Reply:
for me i think the current method is more intuitive; i don't always remember the start of a book title (especially if it starts with "A" or "The") so i feel like being able to search by keywords ("Gatsby", "Two Cities") might save some confusion
happy to discuss more in the slack channel though!